custom_python_actions/check_external_contrib.py

```diff
@@ -1,46 +1,38 @@
-import base64
 import os
 import time
-from typing import Dict, List, NoReturn, Union
+from typing import List
 
-import requests
+import github3
 
 
-def download_file(url: str) -> Union[Dict, NoReturn]:
-
+def download_gh_file(repo: github3.github.repo, file_path: str) -> str:
     # sometimes the request does not work the first time, so set a retry
     for attempt in range(5):
-        x = requests.get(url)
-        data = x.json()
         try:
-            data["content"]
-            data["encoding"]
+            file_content = repo.file_contents(file_path)
             break
-        except KeyError as error:
+        except ConnectionResetError as error:
```

Comment:
Hmm... for now I'm not sure what the other errors would be, so I'd want it to raise the error and then I can add more specific error handling. But this specific error is kind of a fluke, so that's why I had it catch it specifically, so it would retry it 5 times.